### PR TITLE
Support border/outline-radius properties in geckolib.

### PR DESCRIPTION
**Pull request by heycam**
<s>Because of the way Gecko stores outline-width in nsStyleOutline::mOutlineWidth (i.e., as its specified value for keywords) we need separate property implementations for servo/gecko products.</s>

The -moz-outline-radius parsing is a bit of a hack; I was just avoiding the effort of factoring out the parsing into a separate method. Let me know if I should do that.

<s>Gecko-side changes are https://bugzilla.mozilla.org/show_bug.cgi?id=1271168.</s>

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11086)

### CODE REVIEW

**Comment by heycam** [comment]
Per IRC discussion, I pulled out the outline-width changes since they need more work.

File: ports/geckolib/properties.mako.rs [hunk 1]
```diff
@@ -294,6 +294,25 @@ def set_gecko_property(ffi_name, expr):
...
+<%def name="impl_corner_style_coord(ident, x_unit_ffi_name, x_union_ffi_name, y_unit_ffi_name, y_union_ffi_name)">
+    fn set_${ident}(&mut self, v: longhands::${ident}::computed_value::T) {
+        v.0.width.to_gecko_style_coord(&mut self.gecko.${x_unit_ffi_name},
+                                       &mut self.gecko.${x_union_ffi_name});
+        v.0.height.to_gecko_style_coord(&mut self.gecko.${y_unit_ffi_name},
+                                        &mut self.gecko.${y_union_ffi_name});
+    }
+    fn copy_${ident}_from(&mut self, other: &Self) {
+        use gecko_style_structs::nsStyleUnit::eStyleUnit_Calc;
+        assert!(self.gecko.${x_unit_ffi_name} != eStyleUnit_Calc &&
```
Comment (bholley) [inline ports/geckolib/properties.mako.rs]: make this debug_assert!, since assert! is release-mode.

Comment (bholley) [inline ports/geckolib/properties.mako.rs]: And please also fix the one in impl_split_style_coord.

File: components/style/properties/longhand/outline.mako.rs [hunk 2]
```diff
@@ -58,4 +58,10 @@ ${helpers.predefined_type("outline-color", "CSSColor", "::cssparser::Color::Curr
     }
 </%helpers:longhand>
 
+% for corner in ["topleft", "topright", "bottomright", "bottomleft"]:
```
Comment (bholley) [inline components/style/properties/longhand/outline.mako.rs]: I'd add a comment indicating that this property is deprecated and not on a standards track (which is my reading of the situation).

File: components/style/properties/shorthand/outline.mako.rs [hunk 3]
```diff
@@ -47,3 +47,20 @@
...
+)}" products="gecko">
+    use properties::shorthands;
+
+    // Re-use border-radius parsing.
+    shorthands::border_radius::parse_value(context, input).map(|longhands| {
+        Longhands {
+            _moz_outline_radius_topleft: longhands.border_top_left_radius,
+            _moz_outline_radius_topright: longhands.border_top_right_radius,
+            _moz_outline_radius_bottomright: longhands.border_bottom_right_radius,
+            _moz_outline_radius_bottomleft: longhands.border_bottom_left_radius,
```
Comment (bholley) [inline components/style/properties/shorthand/outline.mako.rs]: This can be done with a % for: loop.

**Comment by bholley** [comment]
r=me with comments addressed.

@bors-servo delegate+

**Pull request merged**